apps/opik-frontend/src/components/pages/PromptPage/PromptTab/PromptTab.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import { Button } from "@/components/ui/button";
+import { Info, Pencil } from "lucide-react";
+import { PromptWithLatestVersion } from "@/types/prompts";
+import Loader from "@/components/shared/Loader/Loader";
+import usePromptVersionsById from "@/api/prompts/usePromptVersionsById";
+import UseThisPromptDialog from "@/components/pages/PromptPage/PromptTab/UseThisPromptDialog";
+import EditPromptDialog from "@/components/pages/PromptPage/PromptTab/EditPromptDialog";
+import CommitHistory from "@/components/pages/PromptPage/PromptTab/CommitHistory";
+import usePromptVersionById from "@/api/prompts/usePromptVersionById";
+import { StringParam, useQueryParam } from "use-query-params";
+
+interface PromptTabInterface {
+  prompt?: PromptWithLatestVersion;
+}
+
+const PromptTab = ({ prompt }: PromptTabInterface) => {
+  const [openUseThisPrompt, setOpenUseThisPrompt] = useState(false);
+  const [openEditPrompt, setOpenEditPrompt] = useState(false);
+
+  const [activeVersionId, setActiveVersionId] = useQueryParam(
+    "activeVersionId",
+    StringParam,
+    {
+      updateType: "replaceIn",
+    },
+  );
+
+  const editPromptResetKeyRef = useRef(0);
+
+  const { data } = usePromptVersionsById(
+    {
+      promptId: prompt?.id || "",
+      page: 1,
+      size: 25,
+    },
+    {
+      enabled: !!prompt?.id,
+    },
+  );
+
+  const { data: activeVersion } = usePromptVersionById(
+    {
+      versionId: activeVersionId || "",
+    },
+    {
+      enabled: !!activeVersionId,
+    },
+  );
+
+  const versions = data?.content;
+
+  const handleOpenEditPrompt = (value: boolean) => {
+    editPromptResetKeyRef.current += 1;
+    setOpenEditPrompt(value);
+  };
+
+  useEffect(() => {
+    if (prompt?.latest_version?.id && !activeVersionId) {
+      setActiveVersionId(prompt.latest_version.id);
+    }
+  }, [prompt?.latest_version?.id, activeVersionId, setActiveVersionId]);
+
+  useEffect(() => {
+    return () => {
+      setActiveVersionId(null);
+    };
+  }, [setActiveVersionId]);
+
+  if (!prompt) {
+    return <Loader />;
+  }
+
+  return (
+    <>
+      <div>
+        <div className="flex items-center justify-between w-full">
+          <Button variant="outline" onClick={() => setOpenUseThisPrompt(true)}>
+            <Info className="mr-2 size-4" />
+            Use this prompt
+          </Button>
```

Comment:
Let's hide it until we have the snippets so we can merge without depending on the SDK